cmd/admin-info.go

```
@@ -26,17 +39,130 @@ var adminInfoCmd = cli.Command{
+		var totOffline int
+		var totalOnline int
+		var dispNoOfDisks string
+		for _, disk := range srv.Disks {
+			if disk.State == "ok" {
+				totalOnline++
+			} else {
+				totOffline++
+			}
+		}
+
+		totalDisksPerServer := totalOnline + totOffline
+		totalOnlineDisksCluster += totalOnline
+		totalOfflineDisksCluster += totOffline
+
+		dispNoOfDisks = strconv.Itoa(totalOnline) + "/" + strconv.Itoa(totalDisksPerServer)
+		msg += fmt.Sprintf("   Drives: %s %s\n", dispNoOfDisks, console.Colorize("Info", "OK "))
+
+		msg += "\n"
+	}
+
+	// Summary line on Used space, total number of buckets and objects
+	usedTotal := humanize.IBytes(uint64(u.Usage.Size))
+	msg += fmt.Sprintf("%s Used, %s, %s\n", usedTotal,
+		english.Plural(u.Buckets.Count, "Bucket", ""),
+		english.Plural(u.Objects.Count, "Object", ""))
+	msg += fmt.Sprintf("%s online, %s offline\n",
+		english.Plural(totalOnlineDisksCluster, "drive", ""),
+		english.Plural(totalOfflineDisksCluster, "drive", ""))
+
+	return
+}
+
+// JSON jsonified service status message.
+func (u infoMessage) JSON() string {
+	statusJSONBytes, e := json.MarshalIndent(u, "", "    ")
+	fatalIf(probe.NewError(e), "Unable to marshal into JSON.")
+
+	return string(statusJSONBytes)
+}
+
+// checkAdminInfoSyntax - validate all the passed arguments
+func checkAdminInfoSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) == 0 || len(ctx.Args()) > 1 {
+		cli.ShowCommandHelpAndExit(ctx, "info", 1) // last argument is exit code
+	}
 }
 
-// mainAdminConfig is the handle for "mc admin config" command.
 func mainAdminInfo(ctx *cli.Context) error {
-	cli.ShowCommandHelp(ctx, ctx.Args().First())
+	checkAdminInfoSyntax(ctx)
+
+	// Get the alias parameter from cli
+	args := ctx.Args()
+	aliasedURL := args.Get(0)
+
+	// Create a new MinIO Admin Client
+	client, err := newAdminClient(aliasedURL)
+	fatalIf(err, "Unable to initialize admin connection.")
+
+	// Fetch info of all servers (cluster or single server)
+	admInfo, _ := client.ServerAdminInfo()
```

do not ignore the error here